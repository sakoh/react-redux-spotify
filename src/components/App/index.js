import React, { Component } from 'react'
import './index.css'
import { SearchForm } from '../SearchForm'
import { ArtistsList } from '../ArtistsList'
import { artists } from '../../fixtures'

class App extends Component {
  render () {
    return (
      <div className='App'>
        <SearchForm />
        <ArtistsList title='Search Results' artists={artists} />
      </div>
    )
  }
}

export { App }
