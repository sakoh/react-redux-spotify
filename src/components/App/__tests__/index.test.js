import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import App from '..'

configure({ adapter: new Adapter() })

it('renders without crashing', () => {
  const wrapper = shallow(<App />)
  expect(wrapper).toMatchSnapshot()
})
