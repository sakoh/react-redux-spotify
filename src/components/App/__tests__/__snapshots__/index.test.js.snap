// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="App-header"
>
          <img
                    alt="logo"
                    className="App-logo"
                    src="logo.svg"
          />
          <h1
                    className="App-title"
          >
                     Welcome to Max Artists 
          </h1>
</header>,
        <SearchForm />,
        <ArtistsList
          artists={
                    Array [
                              Object {
                                "image": "http://via.placeholder.com/350x150",
                                "name": "Iron Maiden",
                                "popularity": 10,
                              },
                              Object {
                                "image": "http://via.placeholder.com/350x150",
                                "name": "Metallica",
                                "popularity": 10,
                              },
                              Object {
                                "image": "http://via.placeholder.com/350x150",
                                "name": "Eminenm",
                                "popularity": 10,
                              },
                            ]
          }
          title="Search Results"
/>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="logo"
              className="App-logo"
              src="logo.svg"
/>,
            <h1
              className="App-title"
>
               Welcome to Max Artists 
</h1>,
          ],
          "className": "App-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "className": "App-logo",
              "src": "logo.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Welcome to Max Artists ",
              "className": "App-title",
            },
            "ref": null,
            "rendered": " Welcome to Max Artists ",
            "type": "h1",
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "artists": Array [
            Object {
              "image": "http://via.placeholder.com/350x150",
              "name": "Iron Maiden",
              "popularity": 10,
            },
            Object {
              "image": "http://via.placeholder.com/350x150",
              "name": "Metallica",
              "popularity": 10,
            },
            Object {
              "image": "http://via.placeholder.com/350x150",
              "name": "Eminenm",
              "popularity": 10,
            },
          ],
          "title": "Search Results",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="App-header"
>
            <img
                        alt="logo"
                        className="App-logo"
                        src="logo.svg"
            />
            <h1
                        className="App-title"
            >
                         Welcome to Max Artists 
            </h1>
</header>,
          <SearchForm />,
          <ArtistsList
            artists={
                        Array [
                                    Object {
                                      "image": "http://via.placeholder.com/350x150",
                                      "name": "Iron Maiden",
                                      "popularity": 10,
                                    },
                                    Object {
                                      "image": "http://via.placeholder.com/350x150",
                                      "name": "Metallica",
                                      "popularity": 10,
                                    },
                                    Object {
                                      "image": "http://via.placeholder.com/350x150",
                                      "name": "Eminenm",
                                      "popularity": 10,
                                    },
                                  ]
            }
            title="Search Results"
/>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="logo"
                className="App-logo"
                src="logo.svg"
/>,
              <h1
                className="App-title"
>
                 Welcome to Max Artists 
</h1>,
            ],
            "className": "App-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "logo",
                "className": "App-logo",
                "src": "logo.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " Welcome to Max Artists ",
                "className": "App-title",
              },
              "ref": null,
              "rendered": " Welcome to Max Artists ",
              "type": "h1",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "artists": Array [
              Object {
                "image": "http://via.placeholder.com/350x150",
                "name": "Iron Maiden",
                "popularity": 10,
              },
              Object {
                "image": "http://via.placeholder.com/350x150",
                "name": "Metallica",
                "popularity": 10,
              },
              Object {
                "image": "http://via.placeholder.com/350x150",
                "name": "Eminenm",
                "popularity": 10,
              },
            ],
            "title": "Search Results",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
